syntax = "proto3";
import "google/protobuf/empty.proto";
service Greeter {
  rpc UploadFile(stream UploadFileRequest) returns (StringResponse) {}
  rpc DownloadFile(MetaData) returns (stream FileResponse) {}
  rpc ListFiles(google.protobuf.Empty) returns (FileList) {}
  rpc DeleteFile(MetaData) returns (StringResponse) {}
}

message StringResponse {
  string message = 1;
}
// Download file
message FileResponse{
    bytes chunk_data = 1;
}
message MetaData {
    string filename = 1;
    string extension = 2;
}
message UploadFileRequest {
    oneof request{
        MetaData metadata = 1;
        bytes chunk_data = 2;
    }
}
// List of file on server
message FileList {
    repeated string files = 1;
}